src/ra_utility.h

```diff
@@ -10,12 +10,40 @@
     stated otherwise.
 */
 namespace ra {
+
 template<typename SignedType, class = std::enable_if_t<std::is_signed_v<SignedType>>> _NODISCARD _CONSTANT_FN
 to_unsigned(_In_ SignedType st) noexcept { return static_cast<std::make_unsigned_t<SignedType>>(st); }
 
 template<typename UnsignedType, class = std::enable_if_t<std::is_unsigned_v<UnsignedType>>> _NODISCARD _CONSTANT_FN
 to_signed(_In_ UnsignedType st) noexcept { return static_cast<std::make_signed_t<UnsignedType>>(st); }
 
+/// <summary>
+///   Converts '<paramref name="from" />' into a <typeparamref name="NarrowedType" />.
+/// </summary>
+/// <typeparam name="Narrowed">
+///   A narrower version of <typeparamref name="WideType" />. This template parameter must be
+///   specified.
+/// </typeparam>
+/// <typeparam name="WideType">
+///   The <c>value_type</c> of the input, must have a larger size than
+///   <typeparamref name="NarrowedType" />.
+/// </typeparam>
+/// <param name="from">The arithmetic value to narrowed.</param>
+/// <returns><typeparamref name="WideType" /> as a <typeparamref name="NarrowedType" />.</returns>
+/// <remarks>
+///   This function is used for explicit narrowing conversions when a platform type is different than
+///   a function parameter's type, such as <c>std::size_t</c> and <c>unsigned int</c>.
+/// </remarks>
+template<
+    typename NarrowedType,
+    typename WideType,
+    class = std::enable_if_t<
+    std::is_arithmetic_v<NarrowedType> &&
+    std::is_arithmetic_v<WideType> &&
+    has_smaller_or_same_size_than_v<NarrowedType, WideType>>
+> _NODISCARD _CONSTANT_FN
+narrow_cast(_In_ WideType from) noexcept { return static_cast<NarrowedType>(static_cast<WideType>(from)); }
+
```

Comment:
Even though we currently don't have an x86_64 build, I remembered another comment in different PR about conditional compilation between architectures.

Originally, I was going to make the requirements `ra::has_smaller_size_than_v`, but most platform types are the same as the expected type in `x86 `based processors, they are usually different in `x86_64`. So I made a bit less strict to not contradict the purpose of reducing code-bloat.

This is usually used in `WinAPI `functions where the parameter type is `int `but the input is `size_t`. Would `boxed_cast `be a better name?